public/app/core/services/backend_srv.ts

```diff
@@ -240,34 +244,36 @@ export class BackendSrv {
     });
   }
 
-  createDashboardFolder(name) {
-    const dash = {
-      schemaVersion: 16,
-      title: name.trim(),
-      editable: true,
-      panels: [],
-    };
-
-    return this.post('/api/dashboards/db/', {
-      dashboard: dash,
-      isFolder: true,
-      overwrite: false,
-    }).then(res => {
-      return this.getDashboard('db', res.slug);
-    });
+  createFolder(payload: any) {
+    return this.post('/api/folders', payload);
   }
 
-  saveFolder(dash, options) {
+  updateFolder(folder, options) {
     options = options || {};
 
-    return this.post('/api/dashboards/db/', {
-      dashboard: dash,
-      isFolder: true,
+    return this.put(`/api/folders/${folder.uid}`, {
+      title: folder.title,
+      version: folder.version,
       overwrite: options.overwrite === true,
-      message: options.message || '',
     });
   }
 
+  deleteFolder(uid: string) {
+    let deferred = this.$q.defer();
+
+    this.getFolderByUid(uid).then(folder => {
+      this.delete(`/api/folders/${uid}`)
+        .then(() => {
+          deferred.resolve(folder);
+        })
+        .catch(err => {
+          deferred.reject(err);
+        });
+    });
+
+    return deferred.promise;
+  }
```

Comment:
this function looks a bit strange, should it not be just

```javascript
   deleteFolder(uid: string) {
    return this.delete(`/api/folders/${uid}`)
  }
```

Why is it creating a new promise and fetching the folder before?

Reply:
You're right. It was needed before when we deleted by slug. Will fix

Reply:
@torkelo  My bad. It is needed for the batch delete from manage dashboard page. Kind of the same logic for batch move from the manage dashboard page.